xrpl4j-crypto-parent/xrpl4j-crypto-bouncycastle/src/main/java/org/xrpl/xrpl4j/crypto/bc/signing/DerivedKeyDelegatedSignatureService.java

```diff
@@ -60,11 +61,16 @@
  * requirements, consider a different implementation of {@link DelegatedSignatureService}.</p>
  */
 @SuppressWarnings("UnstableApiUsage")
+// TODO: Make Ed25519KeyPairService and Secp256k1KeyPairService and interface, and move back to core. Then this class
+//  can
+// be moved to core also. Plus, Ed25519KeyPairService should be BcEd25519KeyPairService so that it can coexist with
+// some other type in the future.
```

Comment:
Assuming we want to keep `KeyPairService` instead of just using `WalletFactory`, I don't think having `Ed25519KeyPairService` and `Secp256k1KeyPairService` interfaces would make sense. The key algorithm that a given implementation chooses is an implementation detail. Plus, there would be nothing differentiating the ED interface from the secp256k1 interface, as the contracts currently stand.

+1 though to renaming to `BcEd25519KeyPairService`

Reply:
To add to our conversation from this afternoon, I firmly believe that we should get rid of `KeyPairService` and use `WalletFactory` for everything. In v2, `KeyPairService` also did things like signing and verifying transactions, so it made sense to split `WalletFactory` and `KeyPairService`. Now that the new `KeyPairService` doesn't sign or verify, it basically just does what `WalletFactory` does with different return types and method names...

Although we wanted to have in-memory <--> delegated sibling objects, `KeyPairService` and `DelegatedKeyPairService`, I think this is a fools errand. Contracts with delegated services like cloud KMS providers are so different than in-memory contracts that the only similarity in a lot of these sibling objects is naming. IMO, it is totally fine to have a set of classes/services that are used for in-memory keys -- and this will be the default for 99% of developers -- and a totally different set of classes/services that are for delegated keys.

Reply:
Captured this feedback in #170